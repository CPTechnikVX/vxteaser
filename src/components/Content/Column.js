import React     from 'react';
import Constants from '../../utils/Constants';
import PropTypes from 'prop-types';

/**
 * Column element for the grid
 */
export default class Column extends React.PureComponent {
	static propTypes = {
		/** @ignore */
		children: PropTypes.node,
		width:    PropTypes.number,

	};

	render() {
		const {children, width} = this.props;
		const styleObj          = {
			width: width ? width + '%' : 'auto',
			float: 'left',
		};

		return <div className={Constants.ClassName.Column} style={styleObj}>{children}</div>;
	}
}
