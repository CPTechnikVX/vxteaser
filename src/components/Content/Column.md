```html
<Col width="20%">This is a column of 20% width</Col>
```

```html
<Col>This is a column with an automatic width</Col>
```
