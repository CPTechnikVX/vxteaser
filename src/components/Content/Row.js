import React     from 'react';
import PropTypes from 'prop-types';
import Constants from '../../utils/Constants';

/**
 * Row for the grid
 */
export default class Row extends React.PureComponent {

	static propTypes = {
		/** @ignore */
		children: PropTypes.node,
	};

	render() {
		const {children} = this.props;
		const colCount   = React.Children.count(children);
		const width      = colCount > 0 ? 100 / colCount : 100;

		const childrenWithProps = React.Children.map(
			children,
			(child) => React.cloneElement(child, {
				width: child.props.width ? child.props.width : width,
			})
		);

		return <div className={Constants.ClassName.Row}>{childrenWithProps}</div>;
	}
}
