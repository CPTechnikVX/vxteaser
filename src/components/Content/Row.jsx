import React      from 'react';
import PropTypes  from 'prop-types';
import classnames from 'classnames';
import Constants  from '../../utils/Constants';

/**
 * Row for the grid
 */
export default class Row extends React.PureComponent {
	render() {
		const {children, modifier} = this.props;
		const colCount             = React.Children.count(children);
		const width                = colCount > 0 ? 100 / colCount : 100;
		const classList            = [Constants.ClassName.Row];

		const childrenWithProps = React.Children.map(
			children,
			(child) => React.cloneElement(child, {
				width: child.props.width ? child.props.width : width + '%',
			})
		);

		if (modifier) {
			classList.push(modifier);
		}

		return <div className={classnames(classList)}>{childrenWithProps}</div>;
	}
}

Row.propTypes = {
	/** @ignore */
	children: PropTypes.node,
	modifier: PropTypes.string,
};
