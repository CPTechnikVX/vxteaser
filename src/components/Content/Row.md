```html
<Row>
	<Col>First col of a row</Col>
	<Col>Second col of a row</Col>
	<Col>Third col of a row</Col>
</Row>
```
