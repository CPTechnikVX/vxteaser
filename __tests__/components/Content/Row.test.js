import React     from 'react';
import {shallow} from 'enzyme';
import Row       from '../../../src/components/Content/Row';
import Column    from '../../../src/components/Content/Column';

describe('render row', () => {
	test('with one col', () => {
		const wrapper = shallow(
			<Row><Column>Col1</Column></Row>
		);
		expect(wrapper).toMatchSnapshot();

		expect(wrapper.find('Column').prop('width')).toBe(100);
	});
	test('with two cols', () => {
		const wrapper = shallow(
			<Row><Column>Col1</Column><Column>Col2</Column></Row>
		);
		expect(wrapper).toMatchSnapshot();

		expect(wrapper.find('Column').first().prop('width')).toBe(50);
	});
});
