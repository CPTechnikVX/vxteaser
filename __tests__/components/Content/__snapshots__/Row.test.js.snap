// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render row with one col 1`] = `
<div
  className="vxteaser-row"
>
  <Column
    key=".0"
    width={100}
  >
    Col1
  </Column>
</div>
`;

exports[`render row with two cols 1`] = `
<div
  className="vxteaser-row"
>
  <Column
    key=".0"
    width={50}
  >
    Col1
  </Column>
  <Column
    key=".1"
    width={50}
  >
    Col2
  </Column>
</div>
`;
